.buildkite/pipeline.ml.yml

```
@@ -1,14 +1,44 @@
-- label: ":airplane: ML tests (ray/air)"
+- label: ":airplane: AIR tests (ray/air)"
   conditions: ["NO_WHEELS_REQUIRED", "RAY_CI_ML_AFFECTED"]
   instance_size: medium
   commands:
     - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT
     - DATA_PROCESSING_TESTING=1 INSTALL_HOROVOD=1 ./ci/env/install-dependencies.sh
     - ./ci/env/env_info.sh
-    - bazel test --config=ci $(./ci/run/bazel_export_options) --build_tests_only --test_tag_filters=-gpu,-needs_credentials python/ray/air/...
-    - bazel test --config=ci $(./ci/run/bazel_export_options) --build_tests_only --test_tag_filters=ray_air,-gpu_only,-gpu,-needs_credentials python/ray/train/...
+    - bazel test --config=ci $(./ci/run/bazel_export_options) --build_tests_only --test_tag_filters=-gpu,-needs_credentials
+      python/ray/air/...
     - bazel test --config=ci $(./ci/run/bazel_export_options) --build_tests_only --test_tag_filters=ray_air python/ray/data/...
 
+- label: ":steam_locomotive: Train tests and examples"
+  conditions: ["NO_WHEELS_REQUIRED", "RAY_CI_TRAIN_AFFECTED"]
+  instance_size: large
+  parallelism: 2
+  commands:
+    - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT
+    - TRAIN_TESTING=1 DATA_PROCESSING_TESTING=1 INSTALL_HOROVOD=1 ./ci/env/install-dependencies.sh
+    - ./ci/env/env_info.sh
+    - >-
+      set -x;
+      python ./ci/run/bazel_sharding/bazel_sharding.py
+      --exclude_manual
+      --index "\${BUILDKITE_PARALLEL_JOB}" --count "\${BUILDKITE_PARALLEL_JOB_COUNT}"
+      --tag_filters=-gpu_only,-gpu,-minimal,-tune,-needs_credentials
+      python/ray/train/...
+      > test_shard.txt
+    - cat test_shard.txt
+    - bazel test --config=ci $(./ci/run/bazel_export_options)
+      --test_tag_filters=-gpu_only,-gpu,-minimal,-tune,-needs_credentials
+      $(cat test_shard.txt)
```

Wondering if we can simplify this to

```suggestion
    - python ./ci/run/bazel_sharding/bazel_sharding.py
      --exclude_manual
      --index "\${BUILDKITE_PARALLEL_JOB}" --count "\${BUILDKITE_PARALLEL_JOB_COUNT}"
      --tag_filters=-gpu_only,-gpu,-minimal,-tune,-needs_credentials
      python/ray/train/... | tee /dev/fd/2 | xargs bazel test --config=ci $(./ci/run/bazel_export_options)
      --test_tag_filters=-gpu_only,-gpu,-minimal,-tune,-needs_credentials
```

---

Actually we should probably just have a script for that. We shouldn't need to pass the test tag filters twice

---

@krfricke added the script, PTAL!